### ISSUE TITLE
Error in response codes for Oauth2 token creation

### ISSUE THREAD

**Issue by dcalvoalonso** [ging/fiware-idm#43]
Our authentication process uses the refresh_token grant type to obtain an access_token that can be used to authenticate the request to the Context Broker.

Following the standard (https://tools.ietf.org/html/rfc6749#section-5.2), it seems that error response codes should be HTTP 400 (bad request). Nevertheless, in our tests Keyrock is returning HTTP 500 Internal Server Error. Moreover, the standard states that the error description shall appear in a field called ‘error’ but Keyrock returns this value as part of a field called ``name` together with some additional information (status, code, message, statusCode). It’s curious that the field statusCode is actually 400…

**Comment by apozohue10** [comment]
Fixed!

Sorry for the delay.

**Comment by dcalvoalonso** [comment]
I have just checked this and I think that the responses are not following the standard yet.

For instance:
- If I send a request to get an access token with an invalid `client_secret`. the IDM returns:
  - `Status`: 503. It should be `400`.
  - `Body`: "Server error: missing client `grants`". It should be a JSON payload with:
```
   {
       "error":"invalid_client"
     }
```
- The same happens if I use an invalid `client_id.`
- If I use an invalid `refresh_token` with `grant_type` set to `refresh_token`.
  - `Status`: 400. In this case it is correct
  - `Body`: Invalid grant: refresh token is invalid. It should be a JSON payload with:
```
     {
       "error":"invalid_grant"
     }
```

**Comment by dcalvoalonso** [comment]
This was fixed with commits a599310387af793874ea13b8510be6f567ec4449 to 9bcac9551b30c911618ca4a9a25b68cb8f68031c
